components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  Component,
+  OnInit,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+
+import { FormBuilder, Validators, FormGroup } from '@angular/forms';
+import { Regex } from 'app/helpers/auth/regex';
+
+import { CookbookConstraintGrid } from '../create-environment-modal/create-environment-modal.component';
+
+@Component({
+  selector: 'app-infra-environment-constraint',
+  templateUrl: './infra-environment-constraint.component.html',
+  styleUrls: ['./infra-environment-constraint.component.scss']
+})
+
+export class InfraEnvironmentConstraintComponent implements OnInit {
+
+  @Input() constraintKeys: string[] = [];
+  @Input() cookbookConstraintArray: Array<CookbookConstraintGrid> = [];
```

Comment:
This is both more verbose than it needs to be and also inconsistent. Why should this particular variable include its type in the name ("Array") when all other variables do not? It is generally against best practices because it adds to the maintenance burden (e.g. what if it later became a `List` instead of an `Array`?) 
```suggestion
  @Input() cookbookConstraints: Array<CookbookConstraintGrid> = [];
```

Reply:
Updated the variable name